### PR TITLE
fix router page value inconsistent issue

### LINKED ISSUE #15741: The "Max columns to merge" setting is confusingly displayed
### Affected Version

28.0.1

### Description

In the latest version of the Druid Router page, I noticed that the Compaction task configuration for `Max columns to merge` defaults to `0`. However, the actual default value is `-1` (means unlimited). When switching to the JSON configuration, it shows as `"maxColumnsToMerge": null`. This seems a bit confusing.

### PR DESCRIPTION
**Pull request by fectrain**
Fixes #15741.

#### Fixed the router page value display inconsistent issue

compaction config page
- before
image

- after
image

<hr>

##### Key changed/added classes in this PR
 * `auto-form.tsx`
 * `compaction-config.tsx`

<hr>

This PR has:

- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: web-console/src/components/auto-form/auto-form.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ export interface Field<M> {
   min?: number;
   max?: number;
   zeroMeansUndefined?: boolean;
+  nullMeansUndefined? :boolean
```
Comment (asdf2014) [inline web-console/src/components/auto-form/auto-form.tsx]:
```suggestion
  nullMeansUndefined? :boolean;
```

**Review by vogievetsky — CHANGES_REQUESTED** [review]
Hi, thank you for the fix. I am going to tag it for the next release to make sure it makes it in. I like the direction that you are going here but I have just one comment: It is not aggressive enough! I don't think it makes sense to introduce a flag called `nullMeansUndefined` as I think that flag should be implicitly true by default!!! (maybe we can add a flag called `nullDoesNotMeanUndefined` if we need it later - but we probably wont).

My suggestion is: remove the `nullMeansUndefined` flag that you added and make `toNumberOrUndefined` return undefined on `null` input (instead of `0` - which is wrong). I would add `if (n == null) return;` as the first line of that function. This will likely fix the same sort of issue in other places in the console. 🙏

**Comment by fectrain** [comment]
Hi @vogievetsky , thanks for the review 🙏, the solution your recommended is indeed more precise and universal, updated

**Review by vogievetsky — APPROVED** [review]

File: web-console/src/components/auto-form/auto-form.tsx [hunk 2]
```diff
@@ -84,6 +84,7 @@ export interface Field<M> {
 }
 
 function toNumberOrUndefined(n: unknown): number | undefined {
+  if (n == null) return
```
Comment (vogievetsky) [inline web-console/src/components/auto-form/auto-form.tsx:87]: you need a `;` to comply with the code style. 

This section of the web console readme https://github.com/apache/druid/blob/fb08bac01a20723988b0be31452ce14a87a12207/web-console/README.md#developing talks about how to set up your editor to take care of this for you

Comment (fectrain) [reply web-console/src/components/auto-form/auto-form.tsx]: got it, thanks 🙏 will read it through and make the configuration accordingly

**Review by asdf2014 — APPROVED** [review]

**Pull request merged**